planner/core/rule_join_elimination.go

```diff
@@ -175,13 +176,28 @@ func (o *outerJoinEliminator) isDuplicateAgnosticAgg(p LogicalPlan) (_ bool, col
 
 func (o *outerJoinEliminator) doOptimize(p LogicalPlan, aggCols []*expression.Column, parentSchema *expression.Schema) LogicalPlan {
 	// check the duplicate agnostic aggregate functions
+	childParentSchema := p.Schema()
 	if ok, newCols := o.isDuplicateAgnosticAgg(p); ok {
 		aggCols = newCols
+		childParentSchema = &expression.Schema{
+			Columns: aggCols,
+		}
+	}
+	if logicalSort, ok := p.(*LogicalSort); ok {
```

Comment:
Hmm, I think this is not a general enough fix. What if the the upper node of join is another operator which does not have schema, for example, LogicalTopN or LogicalSelection or other operators we may add in the future?

And the logic here seems incorrect either. What if the upper nodes of LogicalSort uses columns of inner plan? For example:
```sql
create table a(c1 int);
create table b(c1 int primary key);
explain select q from (select a.c1 as p, b.c1 as q from a left join b on a.c1 = b.c1 order by a.c1) as tmp;
```
though for this case using this PR, the final plan contains join which is correct, it mistakenly returns false from function `isParentColsAllFromOuterTable`. This should be another bug I think, we should check if `UniqueID` instead of column name in `isParentColsAllFromOuterTable`.

You can reference `PruneColumns` interface to see what columns does each operator need and use the collected columns from top-down as input of `isParentColsAllFromOuterTable`.

Reply:
In my memory, the same columns from the different plan may have different `UniqueID`. Maybe we should check column name, table name, and database name?

Reply:
For a single column, it may have different column names, but they should have same `UniqueID`. If not, it means we have bugs. @winoros Please confirm this.

Reply:
@lzmhhh123 What's the same column from different plan mean?

Reply:
In this case:
```sql
select t1.a from t t1 left outer join t t2 on t1.a = t2.a;
```
`t1.a` and `t2.a` are different at UniqueID. But I just found that they were also different at `TblName`. So are there any approaches to check the two columns are the same in this case?

Reply:
I think `t1.a` and `t2.a` should be treated as different columns in this case.

Reply:
@lzmhhh123 Currently we cannot distinguish your case easily in fact. And i think we don't need to consider it for now since we have no plan for the near future to integrate tidb with the optimizations related with this case.

Reply:
Yeah. @erjiaqing Pls change all column name check by `UniqueID`.